analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -112,11 +111,11 @@ protected override void Initialize(SonarAnalysisContext context)
 
         protected override void AnalyzeSonar(SyntaxNodeAnalysisContext context, bool isTestProject, bool isScannerRun, SyntaxNode body, ISymbol symbol)
         {
-            var enabledAnalyzers = SonarRules.Where(x => x.SupportedDiagnostics.Any(descriptor => IsEnabled(context, isTestProject, isScannerRun, descriptor))).ToArray();
-            if (!Configuration.ForceSonarCfg) // ToDo: Will be removed in RuleFactory PR
-            {
-                enabledAnalyzers = enabledAnalyzers.Where(x => !x.SupportedDiagnostics.Any(descriptor => AllRules.Keys.Any(dd => descriptor.Id == dd.Id))).ToArray();
-            }
+            var allSonarRules = SonarRules.Union(AllRules
+                                .Select(x => x.Value.CreateSonarFallback(Configuration))
+                                .WhereNotNull()
+                                .Cast<ISymbolicExecutionAnalyzer>()); // Only ISymbolicExecutionAnalyzer should be passed as TSonarFallback to CreateFactory. Have you passed a Roslyn rule instead?
+            var enabledAnalyzers = allSonarRules.Where(x => x.SupportedDiagnostics.Any(descriptor => IsEnabled(context, isTestProject, isScannerRun, descriptor))).ToList();
```

Comment:
Let's do it the other way around to deal with the formatting issue. Calcullate AllRules and union with sonar:
```
            var allSonarRules = AllRules
                .Select(x => x.Value.CreateSonarFallback(Configuration))
                .WhereNotNull()
                .Cast<ISymbolicExecutionAnalyzer>() // Only ISymbolicExecutionAnalyzer should be passed as TSonarFallback to CreateFactory. Have you passed a Roslyn rule instead?
                .Union(SonarRules);
```

Reply:
After that, we can continue with the original `Where` to calculate `enabledAnalyzers` in one go. The original `.ToList()` will need to go to its own line.